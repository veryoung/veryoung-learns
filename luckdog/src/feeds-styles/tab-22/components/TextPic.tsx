/** 左文右图
 * @Author: liquid
 * @Date:   2019年10月15日21:01:50
 *
 */

import React from 'react';
import { View, Image, Text, StyleSheet } from '@tencent/hippy-react-qb';
import FeedsProtect from '../../../mixins/FeedsProtect';
import FeedsTheme from '../../../framework/FeedsTheme';
import { shouldComponentUpdate, countReRender } from '@tencent/luckbox-react-optimize';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  textView: {
    flexDirection: 'column',
    flexWrap: 'wrap',
    flex: 1,
  },
  textMain: {
    fontSize: 16,
    colors: FeedsTheme.SkinColor.A1,
    lineHeight: 24,
    textAlign: 'justify',
  },
  textSub: {
    fontSize: 12,
    colors: FeedsTheme.SkinColor.A3,
    lineHeight: 16,
    marginTop: 4,
  },
  image: {
    marginLeft: 12,
    width: 91,
    height: 68,
    borderRadius: 2,
  },
});

interface Props {
  itemBean: any;
  style: Record<string, any>;
  sMainText: string;
  sSubText: string;
  sImage: string;
  openUrl: (...args) => void;
}

@FeedsProtect.protect
export class TextPic extends React.Component<Props> {
  public constructor(props) {
    super(props);
    this.shouldComponentUpdate = shouldComponentUpdate(this, 'TextPic');
  }

  // {
  //     sMainText: '新生婴儿投胎未饮孟婆汤，竟拥有了上一世的所有记忆。',
  //     sSubText: '选自《堕入星空的少女》',
  //     sImage: 'https://cdn.bookimg.html5.qq.com/pic?picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F274%2F499274%2Ft3_499274.jpg&format=2&flag=2&bookid=1100499274',
  //     sUrl: 'https://bookshelfsz.sparta.html5.qq.com/?t=native#!/bookstore/tuijian/',
  // };
  public render() {
    countReRender(this, 'TextPic');
    const { props } = this;
    const { itemBean = {} } = this.props;
    return (
      <View style={[styles.container, this.props.style]} onClick={props.openUrl}>
        <View style={styles.textView}>
          <View>
            <Text style={styles.textMain} numberOfLines={2}>
              {props.sMainText}
            </Text>
          </View>
          <View>
            <Text style={styles.textSub} numberOfLines={1}>
              {props.sSubText}
            </Text>
          </View>
        </View>
        <View>
          <Image
            style={styles.image}
            source={{ uri: props.sImage }}
            resizeMode='cover'
            reportData={{ sourceFrom: itemBean.item_id }}
          />
        </View>
      </View>
    );
  }
}
