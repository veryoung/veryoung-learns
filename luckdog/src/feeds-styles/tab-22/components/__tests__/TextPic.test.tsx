import React from 'react';
import renderer from 'react-test-renderer';

import { TextPic } from '../TextPic';
import { commonProps } from '@test/mocks/props';

describe('<TextPic/>', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const props = {
        ...commonProps,
        itemBean: {},
        style: {},
        sMainText: '',
        sSubText: '',
        sImage: '',
        openUrl: () => { /*  */ },
      };
      const tree = renderer.create(<TextPic {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
