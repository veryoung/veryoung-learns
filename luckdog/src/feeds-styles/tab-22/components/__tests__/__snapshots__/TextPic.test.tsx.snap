// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextPic/> render testing render normal 1`] = `
<div
  nativeName="View"
  onClick={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {},
    ]
  }
>
  <div
    nativeName="View"
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "flexWrap": "wrap",
      }
    }
  >
    <div
      nativeName="View"
      style={Object {}}
    >
      <p
        nativeName="Text"
        numberOfLines={2}
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "justify",
          }
        }
        text=""
      >
        
      </p>
    </div>
    <div
      nativeName="View"
      style={Object {}}
    >
      <p
        nativeName="Text"
        numberOfLines={1}
        style={
          Object {
            "colors": Array [
              "#8f8f8f",
              "#42464d",
              "#24242460",
              "#ffffff60",
            ],
            "fontSize": 12,
            "lineHeight": 16,
            "marginTop": 4,
          }
        }
        text=""
      >
        
      </p>
    </div>
  </div>
  <div
    nativeName="View"
    style={Object {}}
  >
    <img
      alt=""
      nativeName="Image"
      reportData={
        Object {
          "sourceFrom": undefined,
        }
      }
      resizeMode="cover"
      style={
        Object {
          "borderRadius": 2,
          "height": 68,
          "marginLeft": 12,
          "width": 91,
        }
      }
    />
  </div>
</div>
`;
